File: packages/restapi/src/lib/chat/helpers/aes.ts
import * as CryptoJS from "crypto-js"

export const aesEncrypt = ({ plainText, secretKey }: { plainText: string; secretKey: string }): string => {
  return CryptoJS.AES.encrypt(plainText, secretKey).toString()
}

export const aesDecrypt = ({ cipherText, secretKey }: { cipherText: string; secretKey: string }): string => {
  const bytes = CryptoJS.AES.decrypt(cipherText, secretKey)
  return bytes.toString(CryptoJS.enc.Utf8)
}

export const generateRandomSecret = (length: number): string => {
  let result = ''
  const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
  const charactersLength = characters.length
  for (let i = 0; i < length; i++) {
    result += characters.charAt(Math.floor(Math.random() * charactersLength))
  }

  return result
}

File: packages/push-api-dart/lib/src/chat/src/helper/aes.dart
import 'dart:convert';
import 'dart:math' as math;
import 'package:cryptography/cryptography.dart';

String generateRandomSecret(int length) {
  final random = math.Random.secure();
  final characters =
      'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
  final charactersLength = characters.length;
  final buffer = StringBuffer();
  for (var i = 0; i < length; i++) {
    buffer.write(characters[random.nextInt(charactersLength)]);
  }
  return buffer.toString();
}

// AES 256 bits encryption with pkcs7 padding
Future<String> aesEncrypt(
    {required String plainText, required String secretKey}) async {
  // specifing the AES mode and key bit length
  final algorithm = AesCbc.with256bits(
    macAlgorithm: MacAlgorithm.empty,
  );

  // encrypt
  final secretBox = await algorithm.encryptString(
    plainText,
    secretKey: await algorithm.newSecretKeyFromBytes(utf8.encode(secretKey)),
  );

  // return the cipher text in string form
  return utf8.decode(secretBox.cipherText);
}

aesDecrypt({required String cipherText, required String secretKey}) async {
  // TODO implement aesDecrypt
}
